defold-spine/commonsrc/vertices.cpp

```diff
@@ -35,6 +37,89 @@ static uint32_t EnsureArrayFitsNumber(dmArray<T>& array, uint32_t num_to_add)
     return prev_size;
 }
 
+
+void GetSkeletonBounds(const spSkeleton* skeleton, SpineModelBounds& bounds)
+{
+    dmArray<float> scratch; // scratch buffer
+
+    // a "negative" bounding rectangle for starters
+    bounds.minX = FLT_MAX;
+    bounds.minY = FLT_MAX;
+    bounds.maxX = FLT_MIN;
+    bounds.maxY = FLT_MIN;
+
+    // For each slot in the draw order array of the skeleton
+    for (int s = 0; s < skeleton->slotsCount; ++s)
+    {
+        spSlot* slot = skeleton->drawOrder[s];
+
+        // Fetch the currently active attachment, continue
+        // with the next slot in the draw order if no
+        // attachment is active on the slot
+        spAttachment* attachment = slot->attachment;
+        if (!attachment)
+        {
+            continue;
+        }
+
+        // Fill the vertices array depending on the type of attachment
+        if (attachment->type == SP_ATTACHMENT_REGION)
+        {
+            // Cast to an spRegionAttachment so we can get the rendererObject
+            // and compute the world vertices
+            spRegionAttachment* regionAttachment = (spRegionAttachment*)attachment;
+
+            EnsureArrayFitsNumber(scratch, 4*2);
```

Comment:
We should be able to move this outside of the loop, since we know we'll likely to use this amount of memory.
I suggest putting it on line 44

Reply:
wdym ?

Move `spRegionAttachment* regionAttachment` definition

or

`EnsureArrayFitsNumber(scratch, 4*2);`

Note. scratch has different size for region and mesh type of attachment

Reply:
Correct, I meant "EnsureArrayFitsNumber"
Yes, they're different, but allocating at least 4 vertices before entering the loop is ok, since the scratch buffer is growing. Sure, it is _possible_ that there is a single triangle in the mesh attachment, but allocating 4 vertices beforehand isn't costly.